src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -644,4 +649,46 @@ private List<Long> parseSupplementalGroupList(String gids) {
         }
         return Collections.unmodifiableList(builder);
     }
+
+    /**
+     * <p>
+     * {@link PodDecorator} allowing to inject in {@code jnlp} containers definition a {@code securityContext} definition allowing to use the
+     * {@code restricted} <a href="https://kubernetes.io/docs/concepts/security/pod-security-standards/">Pod Security Standard</a>.
+     * </p>
+     * <p>
+     * See <a href="https://issues.jenkins.io/browse/JENKINS-71639">JENKINS-71639</a> for more details.
+     * </p>
+     */
+    @Extension
+    public static class RestrictedPssSecurityContextInjector implements PodDecorator {
+
+        @NonNull
+        @Override
+        public Pod decorate(@NonNull KubernetesCloud kubernetesCloud, @NonNull Pod pod) {
+            if (kubernetesCloud.isRestrictedPssSecurityContext()) {
+                Optional<Container> maybeJNLP = pod.getSpec().getContainers().stream().filter(container -> JNLP_NAME.equals(container.getName())).findFirst();
```

Comment:
Given

https://github.com/jenkinsci/kubernetes-plugin/blob/746e28c7546b7156692a162927cd52129762d8fb/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java#L305-L311

not having a JNLP container is an error case. You should be throwing `IllegalStateException`

Reply:
:+1: Updated accordingly. thanks!